kernel/arch/dreamcast/hardware/spu.c

```diff
@@ -83,11 +84,25 @@ void spu_memload_sq(uintptr_t dst, void *src_void, size_t length) {
     /* Add in the SPU RAM base (cached area) */
     dst |= SPU_RAM_BASE;
 
+    /* First lock SQs because there is a mutex there. */
+    sq_lock(NULL);
```

Comment:
This isn't necessary, since `sq_cpy()` locks and unlocks the SQs already, right?

Reply:
Yes and no. The thing is that the G2 lock disables IRQs, and the SQ lock uses a mutex.
If SQ mutex is already locked and we try lock it with disabled IRQs then we get deadlock.
If you have a better solution, I'll be glad, because it looks bad. I think I wrote comments to explain it, but apparently they are bad too haha.

Reply:
Lock the SQs before disabling the interrupts.

Reply:
You mean change comment to "Lock the SQs before disabling the interrupts."?